### DISCUSSION TITLE
How to add extra command line parameters to EXISTING backup

### DISCUSSION THREAD

**Discussion by jbrepogmailcom** [borgbase/vorta#811] [Ideas]
**Describe the bug**
There is no possibility to add command line parameters to borg backup when the repository is already created. I want to limit backup to one filesystem (exclude mount points) by -x, but it is nowhere to add.

When trying to add the command to newly linked repository, there is error in log:
borg: error: unrecognized arguments: -x

**To Reproduce**
1. Create a new backup repository
2. Try to add command line parameter -x
3. Check log: borg: error: unrecognized arguments: -x

**Desktop (please complete the following information):**
 - OS: Mac OS Catalina

**Additional context**
If appropriate include logs. Can be found in Main Window > Misc Tab > Log.

**Comment by jbrepogmailcom** [comment]
When I think about the issue it can actually be quite important. Imagine you have encrypted volume, and you mount it decrypted only when user is present. But if no -x parameter is present, that decrypted volume is backed up. Then when the intruder gets access to Vorta GUI, it can mount decrypted backup, because the password for borg decryption key is stored in Vorta settings. I see this as security problem. I recommend that for automated backup the password is used automatically, but for restore, extract and mounting, the password would always have to be entered manually. Plus that optional -x parameter of course. Until that happens, I have to unistall Vorta and use command line.

**Comment by samuel-w** [comment]
-x is used for create only, so this would be more like https://github.com/borgbase/vorta/issues/274.

**Comment by PitBullCH** [comment]
Recently needed to add --remote-path=borg1 to the Borg args for existing Rsync.net repo's that I had - was surprised to see basic editting of this nature was missing, only solution was to delete profile, unlink repo, then new profile, add existing repo and then put the arg into the Borg args field. Very clunky. Should be able to edit the Borg args field (the Edit Profile name dialogue could be extended to allowing editting of the Borg args and maybe other items.

**Comment by samwilson** [comment]
> only solution was to delete profile, unlink repo, then new profile, add existing repo and then put the arg into the Borg args field

Another workaround is to export one profile, edit the JSON file to update the `"extra_borg_arguments": "--remote-path=borg1"`, and then import it to a new profile.

But yeah, being able to edit the advanced options for an existing repo would be great.

(rsync.net now have `borg12` and `borg14` by the way, with `borg1` being an alias of the former.)

**Reply by goebbe** [reply]
Yet another workaround, on Linux, is the use of the "**DB Browser for sqlite"** aka "**sqlitebrowser**" to edit the database (DB) with the Vorta settings directly: 
1. Install the DB Browser
2. Quit Vorta completely
3. Open `~/.local/share/Vorta/settings.db`, using the DB Browser
    If you use the _Flatpak_ build, open:  `~/.var/app/com.borgbase.Vorta/data/Vorta/settings.db`
4. Choose the "Browse data" tab, select the table "repomodel", edit "exta_borg_arguments", Apply
5. Close DB Browser, restart Vorta

Disclaimer: Use with care, since this will edit the SQLite settings.db of Vorta directly. So there is no sanity checking of any kind. 
Just in case: Make a copy of settings.db before editing.   ;-)

**Reply by goebbe** [reply]
> (rsync.net now have `borg12` and `borg14` by the way, with `borg1` being an alias of the former.)

@samwi Did you actually succeed in changing the extra_borg_argument to borg14, by editing the JSON file? 
For some reason, this did not work for me - even when I deleted the old profile before re-importing the profile.json. 
I did not investigate yet, but the comment from @PitBullCH is pointing to a possible reason: The repo does exist independent of the profile - so just re-importing profile settings might not be enough to change extra_borg_arguments (that belong to an existing repo, that might be used elsewhere).

**Comment by goebbe** [comment]
There is an open issue/ enhancement request for editing repository settings: https://github.com/borgbase/vorta/issues/1963
